code/script.js

```diff
@@ -303,41 +388,120 @@ const filterCharacters = (keep) => {
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  }
+  else if (group === 'other') {
     if (keep) {
+      alert(
+        `Yes, the person is ${attribute}! Keep all who ${attribute}`
+      )
+    } else {
+      alert(
+        `No, the person is not a ${attribute}! Remove all who ${attribute}`
+      )
+    }
+  }
+    
+  else if (group === 'hair color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} hair! Keep all who has ${value} hair`
+      )
       // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
     } else {
+      alert(
+        `No, the person does not have ${value} hair! Remove all who has ${value}`
+      )
       // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
     }
   }
+  else if (group === 'eye color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all who has ${value}`
+      )
+    } else {
+      alert(
+        `No, the person does not have ${value} eyes! Remove all who has ${value}`
+      )
+    }
+  }
+  else if (group === 'character') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} character! Keep all who are ${value}`
+      )
+    } else {
+      alert(
+        `No, the person is not ${value}! Remove all who are ${value}`
+      )
+    }
+  }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  if (keep === false) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) 
+  } else {
+  charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
 
+  }
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
+  countingGuesses()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
   // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  confirm("Are you sure?")
+  checkMyGuess(suspect)
 }
 
-// If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (secret.name === suspect) {
+    board.style.display = "none";
+    winOrLoseWrapper.style.display = "flex";
+    winOrLoseText.innerHTML = `
+    <audio autoplay= "autoplay">
+      <source src= "sound/victory.wav" type="audio/mpeg">
+    </audio>
+    <h1 class="glow-won">You have WON !!!</h1>
+  `
```

Comment:
Indentation possible from line 464-468
```suggestion
    winOrLoseText.innerHTML = `
      <audio autoplay= "autoplay">
      <source src= "sound/victory.wav" type="audio/mpeg">
      </audio>
      <h1 class="glow-won">You have WON !!!</h1>
    `
```